### PR TITLE
add persist state on disk. closes #467

### LINKED ISSUE #467: Persist state between app restarts
PR: https://github.com/dat-land/dat-desktop/pull/500

### PR DESCRIPTION
**Pull request by juliangruber**

### CODE REVIEW

File: app/actions/index.js [hunk 1]
```diff
@@ -191,3 +203,12 @@ export const dropFolder = folder => async dispatch => {
   if (!isDirectory) return
   addDat({ path: folder.path })(dispatch)
 }
+
+export const loadFromLocalStorage = () => dispatch => {
+  const blob = localStorage.getItem('state')
+  if (!blob) return
+  const dats = JSON.parse(blob)
+  for (const dat of dats) {
+    addDat(dat)(dispatch)
```
Comment (martinheidegger) [inline app/actions/index.js]: adding to my former point: This should be called something other than dat  in order to have structure <-> name consistency. `datPath`?

File: app/actions/index.js [hunk 2]
```diff
@@ -191,3 +203,12 @@ export const dropFolder = folder => async dispatch => {
   if (!isDirectory) return
   addDat({ path: folder.path })(dispatch)
 }
+
+export const loadFromLocalStorage = () => dispatch => {
+  const blob = localStorage.getItem('state')
+  if (!blob) return
+  const dats = JSON.parse(blob)
+  for (const dat of dats) {
```
Comment (martinheidegger) [inline app/actions/index.js]: adding to my former point: This should be called something other than dat  in order to have structure <-> name consistency. `datPath`?

Comment (aks-) [reply app/actions/index.js]: Hm,  I think `dat` is fine given `key` and `path` are properties on dat?

Comment (martinheidegger) [reply app/actions/index.js]: `dat` is an object created with `dat-node` - no?

Comment (juliangruber) [reply app/actions/index.js]: I agree, renamed it to `datOpt`

**Comment by juliangruber** [comment]
Adding to the point that it's not compatible with previews Dat Desktop yet: Pause/Resume also isn't persisted yet

**Comment by juliangruber** [comment]
@martinheidegger mind giving this another review?

**Comment by martinheidegger** [comment]
@juliangruber I think you saw the points I wanted to fix. I don't have the time to prepare another PR soonish. Do you want to pick some changes and apply them or keep this PR as is?

**Comment by juliangruber** [comment]
tbqh, this seems to be working good enough and can always be refactored later on (we now know the points, as you said). for our current deadline i'd way rather focus on features like the viewers.

**Pull request merged**